### ISSUE TITLE
Unable to publish with pnpm with github releases

### ISSUE THREAD

**Issue by gamertike** [changesets/action#200]
when i try to use `changeset publish` i get the following error:
https://github.com/guildedts/guilded.ts/runs/7480746311?check_suite_focus=true#step:6:63

but when i use `pnpm -r publish` it works fine. Except for the fact that using this **does not** create a github release for me... Is there anyway can fix this?

**Comment by gamertike** [comment]
**Hey NPM users! I have found a solution!**

Changesets cli has a `tag` command which creates tags for your packages. The reason im tellling you this is because with this command, we can use pnpms `pnpm -r publish` command **with github releases**!

in your package.json, update your publish script to the following:
```json
"scripts": {
    "publish": "changeset tag && pnpm -r publish",
    "version": "changeset version"
}
```

now in your work flow add the following:
```yml
-   name: Create Release Pull Request or Publish to NPM
    uses: changesets/action@v1
    with:
        publish: pnpm run publish
        version: pnpm run version
    env:
        GITHUB_TOKEN: ${{ secrets.API_GITHUB_TOKEN }}
        NPM_TOKEN: ${{ secrets.API_NPM_TOKEN }}
```

**Comment by gamertike** [comment]
Reopening because when i use my method, i get an error when the action tries to push tags that already exist and idk how i can check if pnpm published atleast one package.

**Comment by nyaaao** [comment]
@gamertike by default the checkout action doesn't fetch all history, so the changeset action doesn't see existing tags and is trying to create them. this is easily fixed by running checkout action with `fetch-depth: 0`, as noted in their readme https://github.com/actions/checkout#checkout-v3

**Comment by Andarist** [comment]
That being said - we could accept a PR that would check the tags on the remote to remedy this in the Changesets itself. This way it wouldn't matter if you have already fetched all of the history for your repo or not.

**Comment by unional** [comment]
It is also failing for me.
I have `fetch-depth:0`. The error message doesn't show much about what is the actual problem:

```
> @ release /home/runner/work/clibuilder/clibuilder
> pnpm cs publish


> @ cs /home/runner/work/clibuilder/clibuilder
> changeset "publish"

🦋  info npm info clibuilder
🦋  info clibuilder is being published because our local version (8.0.1) has not been published on npm
🦋  info Publishing "clibuilder" at "8.0.1"
🦋  error 
🦋  error packages failed to publish:
🦋  clibuilder@8.0.1
 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.
Error: The process '/home/runner/setup-pnpm/node_modules/.bin/pnpm' failed with exit code 1
    at m._setResult (/home/runner/work/_actions/changesets/action/v1/dist/index.js:102:7[25](https://github.com/unional/clibuilder/runs/8244246928?check_suite_focus=true#step:8:26)8)
    at m.CheckComplete (/home/runner/work/_actions/changesets/action/v1/dist/index.js:102:6686)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/changesets/action/v1/dist/index.js:102:5723)
    at ChildProcess.emit (events.js:314:20)
    at maybeClose (internal/child_process.js:1022:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:[28](https://github.com/unional/clibuilder/runs/8244246928?check_suite_focus=true#step:8:29)7:5)
Error: The process '/home/runner/setup-pnpm/node_modules/.bin/pnpm' failed with exit code 1
```
